app/services/notifications/notifiable_action/send.rb

```diff
@@ -0,0 +1,50 @@
+# send notification about the action ("Published") that happened on a notifiable (Article)
+module Notifications
+  module NotifiableAction
+    class Send
+      # @param notifiable [Article]
+      # @param action [String] for now only "Published"
+      def initialize(notifiable, action = nil)
+        @notifiable = notifiable
+        @action = action
+      end
+
+      delegate :user_data, :article_data, :organization_data, to: Notifications
+
+      def self.call(*args)
+        new(*args).call
+      end
+
+      def call
+        json_data = {
+          user: user_data(notifiable.user),
+          article: article_data(notifiable)
+        }
+        json_data[:organization] = organization_data(notifiable.organization) if notifiable.organization_id
+        notifications = []
+        # followers is an array and not an activerecord object
+        # followers can occasionally be nil because orphaned follows can possibly exist in the db (for now)
+        followers.compact.sort_by(&:updated_at).reverse[0..10_000].each do |follower|
```

Comment:
Orphaned `follows` can exist in the database, though there is a `before_destroy :destroy_follows` callback when a user is destroyed. The user can possibly be removed by `delete` or the user could be destroyed before the callback was introduced.

Reply:
It could be solved by deleting the orphaned follows (I think there is no point in storing them) and adding a foreign key to make sure there are no records left when the user is destroyed.